public/electron.js

```diff
@@ -82,15 +82,35 @@ function initialize(queues) {
   )
 
   mainWindow.webContents.on('did-finish-load', () => mainWindow.show())
-  // mainWindow.webContents.openDevTools()
+
+  // Open devtools if DEV_TOOLS is set and has a value, presumably 1
+  if (process.env.PUBLISH50_DEV_TOOLS) {
+    mainWindow.webContents.openDevTools()
+  }
+
   mainWindow.on('closed', () => {
     mainWindow = null
   })
 }
 
 app.on('ready', async () => {
 
-  globalShortcut.register('CommandOrControl+Q', () => app.quit())
+  globalShortcut.register('CommandOrControl+Q', () => {
+
+    // Prompt user before closing the application if there are active jobs
+    if (Object.keys(queues['queues']['video transcoding']['childPool'].retained).length > 0 ||
+        Object.keys(queues['queues']['image processing']['childPool'].retained).length > 0 ||
+        Object.keys(queues['queues']['metadata']['childPool'].retained).length > 0 ||
+        Object.keys(queues['queues']['youtube']['childPool'].retained).length > 0) {
+      dialog.showMessageBox(
+        {type: 'warning',
+         buttons: ['Cancel', 'Quit'],
+         message: 'Looks like publish50 is currently running some tasks. Quitting publish50 will abort all running tasks.\nAre you sure you want to quit?'},
+         i => {
```

Comment:
Let's surround this `i` with parentheses for consistency:

```
(i) => {
  ...
}
```

Reply:
Is that `\n` reflected in the dialog? If that's the case I think you may use template strings (that are surrounded with backticks) and add an actual newline instead of `\n` which might look nicer.

Reply:
Let's rename `i` to `selectedIndex`, perhaps more descriptive and for consistency with the below?